src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -166,15 +168,24 @@ protected void ProcessUrlAttribute(string attributeName, TagHelperOutput output)
                     }
                     else
                     {
-                        var htmlStringValue = attribute.Value as HtmlString;
-                        if (htmlStringValue != null &&
-                            TryResolveUrl(
-                                htmlStringValue.ToString(),
+                        var htmlContent = attribute.Value as IHtmlContent;
```

Comment:
Changing this is part of #3123 and unrelated to anything else you're doing in this PR.